app/src/main/java/com/example/parkingsystem/mvp/presenter/ReservationPresenter.java

```diff
@@ -0,0 +1,80 @@
+package com.example.parkingsystem.mvp.presenter;
+
+import android.util.Log;
+
+import com.example.parkingsystem.entities.Reservation;
+import com.example.parkingsystem.mvp.model.ReservationModel;
+import com.example.parkingsystem.mvp.view.FragmentReservationView;
+
+import java.util.Date;
+
+public class ReservationPresenter {
+
+
+    private final ReservationModel model;
+    private final FragmentReservationView view;
+
+    public ReservationPresenter(ReservationModel reservationModel, FragmentReservationView reservationView) {
+        this.model = reservationModel;
+        this.view = reservationView;
+    }
+
+    public void startDT() {
+        view.setStartDateTimeDialog();
+    }
+
+    public void endDT() {
+        view.setEndDateTimeDialog();
+    }
+
+    public void onReservationCreationButtonPressed() {
+        try {
```

Comment:
the code within a `try {}` should only be the one that could fail. You can return

example, let's say the following line can throw an IllegalArgumentException:
`int parkingNumber = model.setParkingLotNumber(view.getParkingLotNumberEntered());`

then you should do something like this:

```
int parkingNumber = 0;
try {
    parkingNumber = model.setParkingLotNumber(view.getParkingLotNumberEntered());
} catch (IllegalArgumentException exception) {
    Log.e(ParkingPresenter.class.getSimpleName(), exception.toString());
    view.showInvalidNumber();
    return; //this will make sure no more code is executed on this method
}
//CONTINUE WITH YOUR CODE HERE
```

Reply:
Method updated